### PR TITLE
#qSwfwWJp-Sign-In-to-Phantom

### PR DESCRIPTION
**Pull request by Nkbtemmy**
### **What does this PR do?**
This PR contain sign in router 
### **Description of Task to be completed?**
**.** signIn in App with correct email and password
### **How should this be manually tested?**
. open [swagger](https://phantom-caba-ft-backend-qaodxu.herokuapp.com/phantom-docs) (insteady of using postman) to test if login can be successful OR
**.** clone this repo if you don't have it 
**.** switch to this branch by typing` git checkout origin/ft_backend_sign-In_qSwfwWJp`
. create `dotenv` file with information provided in .env.example
**.** Type in terminal `npm run dev` to start server
**.** type in terminal `npm run migrate:reset` and `npm run seeding`
**.** Open POSTMAN and type in `POST:localhost:5000/api/v1/users/login`
**.** user email of `nkubito@phantom.com` and password of `admin`
### **What are the relevant pivotal tracker stories?**
**.** [qSwfwWJp](https://trello.com/c/qSwfwWJp/8-sign-in-to-phantom)
### **Screenshots (if appropriate)**
Screen Shot 2020-10-05 at 09 59 35

### CODE REVIEW

**Comment by Niyiragira** [comment]
I hope you are working on increasing the coverage

**Comment by Niyiragira** [comment]
Screen Shot 2020-10-07 at 14 59 06
There is some error bro. I don't know if it is only me. @Nkbtemmy @karigirwa @ncutixavier you can check too

**Review by ncutixavier — APPROVED** [review]

**Review by karigirwa — APPROVED** [review]

**Review by Niyiragira — APPROVED** [review]
well done

File: src/app.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import i18n from 'i18n'
 import bodyParser from 'body-parser'
 import swaggerUiExpress from "swagger-ui-express";
 import swaggerDocument from "../swagger.json";
+import userRoute from './routes/userRouters';
 
 const app = express();
 app.use(bodyParser.json());
```
Comment (ceelogre) [inline src/app.js]: `json()` is available with newer `express`. Check to see if you really need `bodyParser` anymore.

File: src/contollers/userSignIn.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        Token:token
+    });
+}
+export default logIn;
+
+
+
+
+
+
```
Comment (ceelogre) [inline src/contollers/userSignIn.js]: Why all these empty lines?

File: src/utils/jwt.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv'
+dotenv.config();
+export const signinToken = (params) =>{
+    const token = jwt.sign(params,process.env.secretKey,{expiresIn:process.env.EXPIRES_IN})
```
Comment (ceelogre) [inline src/utils/jwt.js]: What happens if `EXPIRES_IN` is not defined?

**Review by ceelogre — CHANGES_REQUESTED** [review]
Login with the provided credentials only works in Swagger. Locally, here's the error that I get: 
Capture d’écran 2020-10-12 à 15 00 29

File: src/utils/jwt.js (hunk 3 above)
Comment (Nkbtemmy) [reply src/utils/jwt.js]: It will provide an Error because you are trying to use undefined valiable

File: src/app.js (hunk 1 above)
Comment (Nkbtemmy) [reply src/app.js]: about bodyparse solved; but about screenshoot above ; there is no provision of token secret key provided in dotenv file as shown in .env.example

File: src/contollers/userSignIn.js (hunk 2 above)
Comment (Nkbtemmy) [reply src/contollers/userSignIn.js]: removed

File: src/utils/jwt.js (hunk 3 above)
Comment (ceelogre) [reply src/utils/jwt.js]: You should provide a fix for that.

**Review by ceelogre — APPROVED** [review]

**Pull request merged**